src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,35 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (object == null || object.getClass() != this.getClass()) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        if (width == null && rectangle.width != null
+                || width != null && !width.equals(rectangle.width)) {
+            return false;
+        }
+        if (length == null && rectangle.length != null
+                || length != null && !length.equals(rectangle.length)) {
+            return false;
+        }
+        return color == null && rectangle.color == null
+                || color != null && color.equals(rectangle.color);
```

Comment:
see checklist: While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;).